src/components/photo/PhotoUploader.js

```diff
@@ -0,0 +1,55 @@
+import { equals } from 'ramda'
+import { useEffect, useState } from 'react'
+
+import { addPhoto } from '../../utils/api'
+import { PhotoOrganizer } from './PhotoOrganizer'
+
+export const arePhotosUploaded = (photos) =>
+  photos.every((photo) => !photo.isUploading)
+
+export const PhotoUploader = ({ value, onChange }) => {
+  const [photos, setPhotos] = useState(value)
+
+  // Sync changes from value into state
+  useEffect(() => {
+    if (!equals(photos, value)) {
+      setPhotos(value)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [value])
+
+  // We use a useEffect rather than onChange directly in order to call setPhotos with an updater function
+  // Then propagate changes to photos via onChange
+  useEffect(() => {
+    onChange(photos)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [photos])
+
+  const onOrganizerChange = async (organizerPhotos) => {
+    setPhotos(organizerPhotos.map((photo) => ({ ...photo, file: null })))
+
+    await Promise.all(
+      organizerPhotos.map(async (photo) => {
+        if (photo.file) {
+          const oldId = photo.id
+          const resp = await addPhoto(photo.file)
```

Comment:
There should be a try/catch around this. If photos are not accepted by the API the user just sees them as 'uploading' forever.